consensus/src/detail/tests/processing_tests.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): S. Afach, A. Sinitsyn
+
+use crate::detail::tests::*;
+use blockchain_storage::Store;
+use common::chain::{config::create_mainnet, OutputSpentState};
+
+#[test]
+fn test_process_genesis_block_wrong_block_source() {
+    common::concurrency::model(|| {
+        // Genesis can't be from Peer, test it
+        let config = create_mainnet();
+        let storage = Store::new_empty().unwrap();
+        let mut consensus = Consensus::new_no_genesis(config.clone(), storage).unwrap();
+
+        // process the genesis block
+        let block_source = BlockSource::Peer(0);
+        let result = consensus.process_block(config.genesis_block().clone(), block_source);
+        assert_eq!(result, Err(BlockError::InvalidBlockSource));
+    });
+}
+
+#[test]
+fn test_process_genesis_block() {
+    common::concurrency::model(|| {
+        // This test process only Genesis block
+        let config = create_mainnet();
+        let storage = Store::new_empty().unwrap();
+        let mut consensus = Consensus::new_no_genesis(config, storage).unwrap();
+
+        // process the genesis block
+        let block_source = BlockSource::Local;
+        let block_index = consensus
+            .process_block(consensus.chain_config.genesis_block().clone(), block_source)
```

Comment:
Because it has to be processed just like any other block. It has to be analyzed, its outputs should become available to the UTXO set, and it should be written to the database as a valid block, and should have a valid block index. We can either do the same work twice, or make it a valid block for processing.